src/utils/checkDeviceHasWebcam.ts

```diff
@@ -0,0 +1,45 @@
+export async function checkDeviceHasWebcam(): Promise<boolean> {
```

Comment:
This function has few `if`s that tries to establish the correct way of enumerating the camera devices and save it to `navigator.enumerateDevices`. But there is no such a thing as `navigator.enumerateDevices` so I suggest:
1. Create local variable instead of saving it to `navigator`.
2. Add comments to `if` blocks to document what kind of devices are targeted in those cases.